crate/operator/handlers/handle_update_cratedb.py

```diff
@@ -119,8 +124,18 @@ async def update_cratedb(
                     "size"
                 ) != new_spec.get("resources", {}).get("disk", {}).get("size"):
                     do_expand_volume = True
-
-    if not do_upgrade and not do_restart and not do_scale and not do_expand_volume:
+                elif has_compute_changed(old_spec, new_spec):
+                    do_change_compute = True
+                    # pod resources won't change until each pod is recreated
+                    do_restart = True
+
+    if (
+        not do_upgrade
+        and not do_restart
+        and not do_scale
+        and not do_expand_volume
+        and not do_change_compute
```

Comment:
There's a warning here about usage-before-assign - I think you might need to initialize it above around line 99.